### PR TITLE
feat(design): create `daffArticleEncapsulatedMixin` to prevent article styles from cascading down to nested @daffodil/design components

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Article styles cascade down to design components that are nested inside a `<daff-article>` 
Fixes: N/A

## What is the new behavior?
1. Create `daffArticleEncapsulatedMixin`. Extending the mixin in a component will prevent article styles from cascading down into a component's styles.
2. Implement `daffArticleEncapsulatedMixin`
```
Components that implement the mixin:
Button
Card
Hero
Accordion
Callout
Card
Hero
LinkSet
List
MediaGallery

Components that **do not** implement the mixin:
Article - DNI - i'm me
Backdrop - DNI - irrelevant
Image - DNI - irrelevant
Form - DNI - does not require solo usage
Loading Icon - DNI - irrelevant
Progress Indicator - DNI - irrelevant
Feature - ignored, deprecated
Image Gallery - ignored, deprecated
ImageList - ignored, deprecated
Modal - DNI - irrelevant
Navbar - DNI - low probability of solo usage
Paginator - DNI - low probability of solo usage
QtyDropdown - ignored, deprecated
Sidebar - DNI - low probability of solo usage
```
3. `DesignLandArticleEncapsulatedComponent`'s purpose is to extend this mixin to components that typically have low probability of usage, but is needed for the sake of docs UI.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/design/src/molecules/article/article/article-theme.scss [hunk 1]
```diff
@@ -29,10 +25,14 @@
 			h6 {
 				color: $text-color;
 			}
-		}
 
-		> p {
-			color: $text-color;
+			a {
+				color: daff-color($primary);
```
Comment (damienwebdev) [inline libs/design/src/molecules/article/article/article-theme.scss:30]: I'm not certain on this one. `a` tags can exist inside of many different tags, including `p` tags. How should we accommodate this?

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/design/src/molecules/article/article/article-theme.scss (hunk 1 above)
Comment (xelaint) [reply libs/design/src/molecules/article/article/article-theme.scss:30]: I tried re-enforcing text colors on a[daff-button], but the article style still overrides it. I could add specificity for p > a?

**Comment by damienwebdev** [comment]
@xelaint why can't we wrap the page in an article?

**Comment by xelaint** [comment]
> @xelaint why can't we wrap all design-land pages in an article?

It seems like I still have to import `DaffArticleModule` in the individual pages anyway, despite having it imported in `app.module.ts`

**Comment by xelaint** [comment]
@damienwebdev wrapped all design-land pages in an article

**Comment by xelaint** [comment]
rebase & pushed directly to develop

**Pull request closed without merging**